pkg/cmd/app.go

```diff
@@ -226,17 +228,26 @@ func newOSWatcher(sigs ...os.Signal) chan os.Signal {
 	return sigChan
 }
 
-func action(c *cli.Context) error {
+func action(c *cli.Context) (err error) {
 restart:
 
+	// The purpose of this function is to capture any panic that may occur
+	// during initialization and return an error.
+	defer func() {
+		if r := recover(); r != nil {
+			logrus.WithField("Stack trace", string(debug.Stack())).Error("encountered a failure: %v")
```

Comment:
1. We have constants for logger fields: https://github.com/NVIDIA/dcgm-exporter/blob/main/pkg/dcgmexporter/const.go#L51
2. Please rename the field from "Stack trace" to "stacktrace", for example. It will be hard for ELK to work with fields like: "Stack trace" which include a blank space in the name.

Reply:
3. Today, we start all log messages from capital letters.

Reply:
Thanks for catching this, a mistake on my part. Fixing it.